pkg/catalog/egress.go

```diff
@@ -27,7 +27,8 @@ const (
 
 // GetEgressTrafficPolicy returns the Egress traffic policy associated with the given service identity
 func (mc *MeshCatalog) GetEgressTrafficPolicy(serviceIdentity identity.ServiceIdentity) (*trafficpolicy.EgressTrafficPolicy, error) {
-	if !mc.configurator.GetFeatureFlags().EnableEgressPolicy {
+	if mc.configurator.GetMeshConfig().Spec.Traffic.EnableEgress {
```

Comment:
super nit: Slightly of the opinion we should just wrap the call of `GetEgressTrafficPolicy` within that check instead of placing it here. Removes the awkward `return nil, nil`

Reply:
I prefer this approach, instead of having a bunch of conditionals on the caller side wherever this function is invoked. A `nil` policy is a no-op for the XDS code generation path.